Firestore/core/src/firebase/firestore/util/executor_std.cc

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/util/executor_std.h"
+
+#include <future>  // NOLINT(build/c++11)
+
+namespace firebase {
+namespace firestore {
+namespace util {
+namespace internal {
+
+namespace {
+
+// The only guarantee is that different `thread_id`s will produce different
+// values.
```

Comment:
If this is now only used in CurrentExecutorName(), you could inline it. If you do so, and also somehow mark that CurrentExecutorName() is only used for debugging purposes, then using hash seems like it *might* be ok. But you just know that we'll eventually get a collision resulting in an impossible to debug state, so I'd still prefer to avoid it.

But how about this: If it's only used for debugging, then surround the entire method definition with an `#if (DEBUG)`? But that's a little awkward, since you could no longer debug thread id related issues using prod builds, so I'm not a big fan of that either.

A hash table of threadid->auto_ids lookups? (Might work, but requires maintaining extra state for not a great reason.)

I don't like any of these options. I'd keep sstream for now; when we find out that our sdk is too large, a simple grep for sstream should remind us of this. (Premature optimization, etc.) But feel free to go a different direction.

Reply:
I used "debugging" a little too loosely. `CurrentExecutorName` is used in `FIREBASE_ASSERT_MESSAGE`s in `AsyncQueue`, so it could show up in a user-reported issue. Agree that a map of ids would work, but at this point it seems really overkill. I'm inclined to leave it as is and revisit if necessary (like you said, it's easily greppable).